File: 100_.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if((p==null)||(q==null))
            return ((p==null)&&(q==null));
        else{
            if (p.val==q.val) // here should pay attention that the value should be equal not the node.
            return isSameTree(p.left, q.left)&&isSameTree(p.right, q.right);
            else
            return false;
        }
    }
}

File: python_code/100_.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isSameTree(self, p, q):
        """
        :type p: TreeNode
        :type q: TreeNode
        :rtype: bool
        """
        A= p
        B= q
        if (A==None)&(B==None):
            return True
            
        
        if A!=None and B!=None :
            return (A.val==B.val)&self.isSameTree(A.left,B.left)&self.isSameTree(A.right,B.right)
        else:       
            return False